InsertParagraph should insert a BR when the root editable element is a phrasing content

rniwa opened the pull request:
#### 37c9962af2e3af8339a71539efdcd40c813d6bd6
<pre>
InsertParagraph should insert a BR when the root editable element is a phrasing content
<a href="https://bugs.webkit.org/show_bug.cgi?id=252219">https://bugs.webkit.org/show_bug.cgi?id=252219</a>

Reviewed by Wenson Hsieh.

When the root editable element only allows phrasing content (e.g. span) then execCommand(&apos;insertParagraph&apos;) should insert a br instead of a div.

* LayoutTests/editing/inserting/insert-paragraph-separator-phrasing-editable-root-expected.txt: Added.
* LayoutTests/editing/inserting/insert-paragraph-separator-phrasing-editable-root.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&amp;display=block&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&amp;display=block&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&amp;display=list-item&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&amp;display=list-item&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&amp;display=block&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&amp;display=block&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&amp;display=list-item&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&amp;display=list-item&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&amp;display=block&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&amp;display=block&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&amp;display=list-item&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&amp;display=list-item&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&amp;display=block&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&amp;display=block&amp;method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&amp;display=list-item&amp;method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&amp;display=list-item&amp;method=shift-enter-expected.txt:
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::isPhrasingContent):
(WebCore::isEditableRootPhrasingContent):
(WebCore::InsertParagraphSeparatorCommand::doApply):

Canonical link: <a href="https://commits.webkit.org/260242@main">https://commits.webkit.org/260242@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7195bf0afd19219cd3ab7adfe385bb9ae9457203

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107639 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16690 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40534 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116786 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116192 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111531 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18112 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8007 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99809 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113394 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13671 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96870 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41341 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95584 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28499 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83181 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9677 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29848 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10352 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6741 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15828 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49433 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11912 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3841 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
```diff
@@ -146,6 +147,83 @@ Ref<Element> InsertParagraphSeparatorCommand::cloneHierarchyUnderNewBlock(const
+    auto* ediableRoot = highestEditableRoot(position);
+    if (!ediableRoot)
+        return false;
+    return enclosingNodeOfType(firstPositionInOrBeforeNode(ediableRoot), isPhrasingContent);
```
```suggestion
    auto* editableRoot = highestEditableRoot(position);
    if (!editableRoot)
        return false;
    return enclosingNodeOfType(firstPositionInOrBeforeNode(editableRoot), isPhrasingContent);
```

rniwa committed:
InsertParagraph should insert a BR when the root editable element is a phrasing content
https://bugs.webkit.org/show_bug.cgi?id=252219

Reviewed by Wenson Hsieh.

When the root editable element only allows phrasing content (e.g. span) then execCommand('insertParagraph') should insert a br instead of a div.

* LayoutTests/editing/inserting/insert-paragraph-separator-phrasing-editable-root-expected.txt: Added.
* LayoutTests/editing/inserting/insert-paragraph-separator-phrasing-editable-root.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&display=block&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&display=block&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&display=list-item&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=normal&display=list-item&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&display=block&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&display=block&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&display=list-item&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre&display=list-item&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&display=block&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&display=block&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&display=list-item&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-line&display=list-item&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&display=block&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&display=block&method=shift-enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&display=list-item&method=enter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/insertparagraph-in-inline-editing-host.tentative_white-space=pre-wrap&display=list-item&method=shift-enter-expected.txt:
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::isPhrasingContent):
(WebCore::isEditableRootPhrasingContent):
(WebCore::InsertParagraphSeparatorCommand::doApply):

Canonical link: https://commits.webkit.org/260242@main
